src/drawers/drawer.ts

```diff
@@ -67,8 +69,17 @@ export module Drawers {
      *
      * @param{any[]} data The data to be drawn
      */
-    protected _enterData(data: any[]) {
-      // no-op
+    private _setSelectionData(data: any[]) {
+      var dataElements = this._selection().data(data);
+      dataElements.enter().append(this._svgElementName);
+      dataElements.exit().remove();
+      this._setDefaultAttributes(dataElements);
+    }
+
+    protected _setDefaultAttributes(selection: d3.Selection<any>) {
```

Comment:
Similarly, "`applyDefaultAttributes()`" might be more appropriate than using "set", which is probably better reserved for simple setters.